litestar/_multipart.py

```diff
@@ -113,51 +68,55 @@ def parse_multipart_form(
-                file_name = form_parameters.get("filename")
-
-                if file_name is None and (filename_with_asterisk := form_parameters.get("filename*")):
-                    encoding, _, value = decode_rfc2231(filename_with_asterisk)
-                    file_name = unquote(value, encoding=encoding or content_charset)
-
-            elif form_header_field == "content-type":
-                content_type = form_header_value
-                content_charset = form_parameters.get("charset", "utf-8")
-            headers.append((form_header_field, form_header_value))
-
-        if field_name:
-            post_data = form_part[line_index:-4].lstrip(b"\r\n")
-            if file_name:
-                form_file = UploadFile(
-                    content_type=content_type, filename=file_name, file_data=post_data, headers=dict(headers)
-                )
-                fields[field_name].append(form_file)
-            elif post_data:
-                fields[field_name].append(post_data.decode(content_charset))
-            else:
-                fields[field_name].append(None)
+    chunk = await async_next(stream, b"")
+    if not chunk:
+        return fields
+
+    try:
+        with PushMultipartParser(boundary, max_segment_count=multipart_form_part_limit) as parser:
+            segment: MultipartSegment | None = None
+            data: UploadFile | bytes = b""
+            while not parser.closed:
+                for form_part in parser.parse(chunk):
+                    if isinstance(form_part, MultipartSegment):
+                        segment = form_part
+                        if segment.filename:
+                            data = UploadFile(
+                                content_type=segment.content_type or "text/plain",
+                                filename=segment.filename,
+                                headers=dict(segment.headerlist),
+                            )
+                    elif form_part:
+                        if isinstance(data, UploadFile):
+                            await data.write(form_part)
+                        else:
+                            data += form_part
+                    else:
+                        # end of part
+                        if segment is None:
+                            # we have reached the end of a segment before we have
+                            # received a complete header segment
+                            raise ClientException("Unexpected eof in multipart/form-data")
+                        if segment.name:
+                            if isinstance(data, UploadFile):
+                                await data.seek(0)
+                                fields[segment.name].append(data)
+                            elif data:
+                                fields[segment.name].append(data.decode(segment.charset or "utf-8"))
+                            else:
+                                fields[segment.name].append(None)
```

Comment:
Shouldn't this be an empty string instead? A text form field without content is empty, but not missing or invalid.

Reply:
We use `None` here for the benefit of the validation libraries consuming this data, which do make an effort to distinguish between `None`, `""` and not being provided a value. In this case, *we* provide the field with a value, but want to signal that no value was provided to us.